File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root) {
        console.log('root: ', root);

        if (!root) {
            return 0;
        }

        const queue = [];
        let depth = 0;
        queue.push(root);
        while (queue.length > 0) {
            const size = queue.length;

            for (let i = 0; i < size; i++) {
                const node = queue.shift();
                if (node.left) queue.push(node.left);
                if (node.right) queue.push(node.right);
            }
            depth += 1;
            // const node = queue.shift();
            // if (node.left) {
            //     queue.push(node.left);
            // }
            // if (node.right) {
            //     queue.push(node.right);
            // }
            // if (node.right || node.left) {
            //     depth += 1;
            // }
        }
    return depth;
    }
}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        
        output = []
        def dfs(node):
            if node.left is None and node.right is None:
                return node.val

            dfs(node.left)
            dfs(node.right)

        return output